requirements/prod_lock.txt

```diff
@@ -37,8 +37,12 @@ fonttools==3.15.0
 gitdb==0.6.4
 google-api-python-client==1.6.2
 goslate==1.5.1            # via pydictionary
+h2==2.6.2                 # via hyper
+hpack==3.0.0              # via h2
 html2text==2016.9.19
 httplib2==0.10.3
+hyper==0.7.0              # via apns2
+hyperframe==3.2.0         # via h2, hyper
 idna==2.6
 ijson==2.3
 ipython-genutils==0.2.0   # via traitlets
```

Comment:
Why didn't this change `dev_lock.txt` as well?

Reply:
Because I made the change manually, and didn't think about that file. :-)

The reason I did it manually is that for me the `pip-compile` command makes some extra changes at least in `prod_lock.txt`:
```
--- a/requirements/prod_lock.txt
+++ b/requirements/prod_lock.txt
@@ -4,11 +4,11 @@
 #
 #    pip-compile --output-file requirements/prod_lock.txt requirements/prod.txt
 #
-git+https://github.com/rafaelmartins/pyoembed.git@eb9901917c2a44b49e2887c077ead84a722c50dc#egg=pyoembed
-git+https://github.com/umairwaheed/talon.git@7d8bdc4dbcfcc5a73298747293b99fe53da55315#egg=talon==1.2.11
-git+https://github.com/umairwaheed/virtualenv-clone.git@short-version#egg=virtualenv-clone==0.2.6
-git+https://github.com/zulip/python-zulip-api.git@0.3.2#egg=zulip==0.3.2&subdirectory=zulip
-git+https://github.com/zulip/python-zulip-api.git@0.3.2#egg=zulip_bots==0.3.2&subdirectory=zulip_bots
+-e git+https://github.com/rafaelmartins/pyoembed.git@eb9901917c2a44b49e2887c077ead84a722c50dc#egg=pyoembed
+-e git+https://github.com/umairwaheed/talon.git@7d8bdc4dbcfcc5a73298747293b99fe53da55315#egg=talon==1.2.11
+-e git+https://github.com/umairwaheed/virtualenv-clone.git@short-version#egg=virtualenv-clone==0.2.6
+-e git+https://github.com/zulip/python-zulip-api.git@0.3.2#egg=zulip==0.3.2&subdirectory=zulip
+-e git+https://github.com/zulip/python-zulip-api.git@0.3.2#egg=zulip_bots==0.3.2&subdirectory=zulip_bots
 apns2==0.3.0
 argon2-cffi==16.3.0
 asn1crypto==0.22.0        # via cryptography
@@ -34,6 +34,8 @@ django-webpack-loader==0.5.0
 django==1.11.4
 docopt==0.6.2
 fonttools==3.15.0
+future==0.16.0            # via python-twitter
+futures==3.1.1            # via django-pipeline, goslate
 gitdb==0.6.4
 google-api-python-client==1.6.2
 goslate==1.5.1            # via pydictionary
```
Clearly something to debug, of course.

Reply:
OK, fixed this by using the somewhat-cleaner, still-manual procedure of
* run the prescribed `pip-compile` command, then
* manually edit extraneous changes back to the status quo.

Reply:
The `-e` stuff is I think what happens if it crashes partway through; if you read the script, it adds and then removes those.  A complicated mess basically related to the fact that `pip-compile` doesn't support `Git` URLs without `-e` or something.  

`future` and `futures` I'm less sure about, but if I had to guess, it's because 3.4 vs. 3.5 vs 3.6 have slightly different Python dependencies, which I noticed for one package before, shortly before we deleted said dependency.   I was hoping to avoid needing one copy per Python version, but maybe we will need to go that route...